syntax = "proto2";

import "protobrain/proto/encoder.proto";
import "protobrain/proto/sdr.proto";

package protobrain.proto;

message Snapshot {
  optional CorticalSnapshot cortex = 1;
  optional SensorSnapshot sensor = 2;
}

message CorticalSnapshot {
  repeated SparseDistributedRepresentation sdr = 1;
}

message SensorSnapshot {
  optional SensorValue value = 1;
  optional SparseDistributedRepresentation sdr = 2;
}
